unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/Common/BIWSearchBarController.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL;
+using UnityEngine;
+
+public interface IBIWSearchBarController
+{
+    bool IsFilterActive();
+    void Dispose();
+    List<Dictionary<string, List<CatalogItem>>> FilterAssets(string nameToFilter);
+}
+
+public class BIWSearchBarController : IBIWSearchBarController
+{
+    public event Action<List<Dictionary<string, List<CatalogItem>>>> OnFilterChange;
+    public event Action OnFilterRemove;
+    internal List<Dictionary<string, List<CatalogItem>>> filterObjects = new List<Dictionary<string, List<CatalogItem>>>();
+
+    private IBIWSearchBarView view;
+
+    private bool isSmartItemFilterActive = false;
+    private bool isFilterActive = false;
+    public BIWSearchBarController(ISceneCatalogView catalogView)
```

Comment:
This is a completely fine way of implementing dependency injection but it can raise some issues with cycle dependencies.

To avoid this we are trying to implement DI with an `Initialize` method. Would it be difficult to do it here as well?